docs/add-builtin-service.md

```diff
@@ -0,0 +1,204 @@
+The `broker.go` file inherits [BrokerBase](../pkg/providers/builtin/base/broker_base.go) and parses the variables given in the Service Definition to make direct API calls to provision or bind services.
+The `registry.go` file lists all built-in service providers. You will need to edit this file to include the new service.
+
+#### Useful Resources
+1. Google's **Go Reference** for the service. 
+
+   To add a built-in service, you will essentially write a service template based on direct Go API calls. Therefore, it's important to be familiar with the Go API.
+   Pay particular attention to the `CreateInstance` function and any required arguments. In the `Provision` function in `broker.go`, you will call `CreateInstance` to provision your new service. 
+
+2. Any **Pricing Tiers**.
+
+   Some resources will have different plan tiers or pricing tiers. 
+   The built-in service may want to offer multiple plans to map to the different pricing tiers. For example, Redis has two [pricing tiers](https://cloud.google.com/memorystore/pricing), so the built-in Redis service offers two plans (BASIC and STANDARD HIGH AVAILABILITY) that map to the two pricing tiers.
+
+3. Google **Documentation** for your new service.
+
+   This will help you fill out fields in the `ServiceDefinition`. The Redis docs can be found [here](https://cloud.google.com/memorystore/docs/redis).
+    
+## Code Walkthrough
+To get an idea of how to create the service template for your new service, I will walk through the changes made to `definition.go`, `broker.go`, and `registry.go` when implementing Redis as a new built-in service. At the end of the code walkthrough, you will also be able to test your new service!
+
+### Service Definition
+
+[Redis example: `../pkg/providers/builtin/redis/definition.go`](../pkg/providers/builtin/redis/definition.go)
+
+The `ServiceDefinition` holds the necessary details to describe an OSB service and provision it. It is defined in [this struct](../pkg/broker/service_definition.go). Some of the basic fields are described below:
+
+**Service Definition Basics**
+* `ID`: Required: A unique Version 4 UUID. You can generate a UUID using the `uuidgen` CLI tool. If that's not installed, you can use this [Online UUID Generator](https://www.uuidgenerator.net/).
+* `Name`: Required: A unique lowercase, hyphen-separated name for the service.
+* `Description`: A description of the service can be found in the Google documentation.
+* `DisplayName`: A human-friendly display name for the service.
+* `ImageUrl`: The url of the `.svg` image of the service icon. This can often be found by inspecting the HTML element in the Google Cloud product page.
+* `DocumentationUrl`: The url of the Google Cloud documentation.
+* `SupportUrl`: The url of the Google Cloud support page. This can often be found through the Google Cloud documentation.
+* `Tags`: Tags to help find the service.
+* `Bindable`: Boolean value.
```

Comment:
nit: what are Bindable and PlanUpdateable for?

Bindable is whether or not the service supports the bind call and planUpdateable should always be false because the GCP service broker doesn't support updating plans